sqldelight-idea-plugin/src/main/kotlin/app/cash/sqldelight/intellij/run/ConnectionManager.kt

```diff
@@ -0,0 +1,47 @@
+package app.cash.sqldelight.intellij.run
+
+import app.cash.sqldelight.intellij.util.dialectPreset
+import app.cash.sqldelight.intellij.util.isSqlite
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.project.Project
+import java.sql.Connection
+import java.sql.DriverManager
+import java.sql.SQLException
+
+internal interface ConnectionManager {
+  fun getConnection(): Connection
+
+  companion object {
+    fun getInstance(project: Project): ConnectionManager {
+      return ServiceManager.getService(project, ConnectionManager::class.java)!!
+    }
+  }
+}
+
+internal class ConnectionManagerImpl(private val project: Project) : ConnectionManager {
```

Comment:
It looks like this is actually the only thing thats dialect-specific. I don't know if its worth promoting into the dialect api but at least for testing we could try bundling like the postgres jdbc artifact and prompting the user for the connection url. Doesnt have to be in this PR, just spitballing